### PR TITLE
Create a workflow that can be triggered on forks

### LINKED ISSUE #7: [public api] add tests to github workflow
add tests to github workflow

### PR DESCRIPTION
**Pull request by RusticPotatoes**
Stops workflows in forked branches from running on selfhosted 
will use public github actions if forked

### CODE REVIEW

File: .github/workflows/pytest-workflow.yml [hunk 1]
```diff
@@ -6,9 +6,10 @@ permissions:
   contents: read
 
 jobs:
-  linter:
-    name: Black Syntax Check
+  Linter-Known-Owner:
+    name: Black Syntax Check on Self Hosted
     runs-on: [self-hosted, "hetzner"]
+    if: github.repository_owner == 'Bot-detector'
```
Comment (extreme4all) [inline .github/workflows/pytest-workflow.yml:20]: can we only add this

File: .github/workflows/pytest-workflow.yml [hunk 2]
```diff
@@ -21,9 +22,10 @@ jobs:
           options: "--check --verbose"
           src: "./src"
 
-  Build-Test:
-    name: Setup Enviornment and Run Tests
+  Build-Test-Known-Owner:
+    name: Setup and Run Tests on Self Hosted
     runs-on: [self-hosted, "hetzner"]
+    if: github.repository_owner == 'Bot-detector'
```
Comment (extreme4all) [inline .github/workflows/pytest-workflow.yml:36]: can we only add this

**Comment by extreme4all** [comment]
Please add a seperate workflow file that can be triggered manual, not automatic where the owner is not bot detector?

**Comment by RusticPotatoes** [comment]
> Please add a seperate workflow file that can be triggered manual, not automatic where the owner is not bot detector?

ok i have separated the manual trigger public runner job

**Pull request merged**